### ISSUE TITLE
Is there a way to report bedmap results in order of IDs or filenames?

### ISSUE THREAD

**Issue by alexpreynolds** [bedops/bedops#198]
Hi Shane,

Let's say I have three BED5 files, `A.bed`, `B.bed`, and `C.bed`:

```
$ more A.bed
chr1    10468   10470   setA    0.97
chr1    10470   10472   setA    1
chr1    10483   10485   setA    0.95

$ more B.bed
chr1    10468   10470   setB    0.79
chr1    10470   10472   setB    0.93
chr1    10483   10485   setB    0.88

$ more C.bed
chr1    10468   10470   setC    0.91
chr1    10470   10472   setC    0.97
chr1    10483   10485   setC    0.87
```

I make a partition and a union of these three sets:

```
$ bedops --partition A.bed B.bed C.bed > partition.bed
$ bedops --everything A.bed B.bed C.bed > union.bed
```

I then map the scores from `union.bed` to the elements in `partition.bed`:

```
$ bedmap --echo --echo-map-score --delim '\t' --multidelim '\t' --prec 2 partition.bed union.bed
chr1	10468	10470	0.97	0.79	0.91
chr1	10470	10472	0.97	1.00	0.93
chr1	10483	10485	0.95	0.88	0.87
```

The score results are not always in the order I'd expect to see.

What is happening in `chr1	10468	10470` seems to put the scores in correct order from `A.bed` — `0.97`, from `B.bed` — `0.79`, and from `C.bed` — `0.91`.

But what gets mapped from `chr1    10470   10472` seems out of order. Results are reported from `C.bed` — `0.97`, from `A.bed` — `1`, and then from `B.bed` — `0.93`.

One way I thought about solving this is to merge the ID and score fields and do post-processing to split the merged field and re-sort.

Is there a way to have `bedmap` use the lexicographical ordering or ID fields, or the order of input files, to have map results come back in a predictable order?

**Comment by sjneph** [comment]
It is supposed to behave in the way that you expect, and this is definitely a bug.  Thanks for the report.

**Comment by sjneph** [comment]
As far as a bad workaround for now, you could use --echo-map and parse things.

**Comment by alexpreynolds** [comment]
No worries, and thanks!

**Comment by sjneph** [comment]
Pushed a proposed fix in d60dc5d956b8ef31593406b86d9ffe8ef68ef1d8 .  I'd be grateful if you'd help kick the tires on that.

**Comment by alexpreynolds** [comment]
Looks good!
